service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -308,6 +313,18 @@ public ResponseEntity<MethodListResponse> getMethods(
     return ResponseEntity.ok(new MethodListResponse().methods(methodDetails));
   }
 
+  public ResponseEntity<ArchiveMethodResponse> archiveMethod(UUID methodId) {
+    // extract bearer token from request to pass down to API calls
+    BearerToken userToken = bearerTokenFactory.from(httpServletRequest);
+
+    // check if current user has write permissions on the workspace
+    if (!samService.hasWritePermission(userToken)) {
```

Comment:
Thinking-out-loud suggestion: Can we do a partial function to avoid this sensitive and surely much-repeated pattern?
e.g.
```
withWritePermission {
    ...do the thing...
}